Dockerfile

```diff
@@ -51,17 +51,34 @@ RUN sed -ri 's,(IGB_UIO=).*,\1n,' config/common_linux* && \
     make config T=$RTE_TARGET && \
     make $MAKEFLAGS EXTRA_CFLAGS="-march=$CPU -g -w -fPIC -DALLOW_EXPERIMENTAL_API"
 
-WORKDIR /
+# Prepare BESS source & apply patches
 ARG BESS_COMMIT=master
+WORKDIR /
 RUN apt-get update && apt-get install -y wget unzip ca-certificates git
 RUN wget -qO bess.zip https://github.com/NetSys/bess/archive/${BESS_COMMIT}.zip && unzip bess.zip
 WORKDIR bess-${BESS_COMMIT}
 COPY core/ core/
 COPY patches/bess patches
 RUN cp -a ${DPDK_DIR} deps/dpdk-19.11.1 && \
-    cat patches/* | patch -p1
-RUN ./build.py --plugin sample_plugin bess && \
-    cp bin/bessd /bin && \
+    cat patches/* | patch -p1 && \
+    mkdir plugins
+
+# Plugins plugins
+## SequentialUpdate plugin
+RUN cp -r sample_plugin plugins
+
+## Network Token plugin
+ARG ENABLE_NTF
+ARG NTF_COMMIT=master
+COPY install_ntf.sh .
+RUN ./install_ntf.sh
+
+# Main BESS build
+COPY build_bess.sh .
+RUN ./build_bess.sh
```

Comment:
```suggestion
./build.py bess --plugin $(ls plugins)
```

Reply:
can combine with below RUN

Reply:
Does this work?  I tested with both sample_plugin and ntf and it complained.

Reply:
I committed the below suggestion but it broke.

Reply:
nope hadnt tested, was off vpn. didnt realise it was not a list `--plugin` arg. I did not see much was moved to the script and suggested that.

Reply:
```diff
diff --git a/Dockerfile b/Dockerfile
index d61ae959..4e0e12c3 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -65,7 +65,7 @@ RUN cp -a ${DPDK_DIR} deps/dpdk-19.11.1 && \
 
 # Plugins plugins
 ## SequentialUpdate plugin
-RUN cp -r sample_plugin plugins
+RUN mv sample_plugin plugins
 
 ## Network Token plugin
 ARG ENABLE_NTF
@@ -73,12 +73,10 @@ ARG NTF_COMMIT=master
 COPY install_ntf.sh .
 RUN ./install_ntf.sh
 
-# Main BESS build
+# Build and copy artifacts
 COPY build_bess.sh .
-RUN ./build_bess.sh
-
-# Copy artifacts
-RUN cp bin/bessd /bin && \
+RUN ./build_bess.sh && \
+    cp bin/bessd /bin && \
     mkdir -p /bin/modules && \
     cp core/modules/*.so /bin/modules && \
     mkdir -p /opt/bess && \
diff --git a/build_bess.sh b/build_bess.sh
index 83bb22cb..a8d71105 100755
--- a/build_bess.sh
+++ b/build_bess.sh
@@ -1,25 +1,21 @@
 #!/usr/bin/env bash
 # SPDX-License-Identifier: Apache-2.0
-# Copyright (c) 2019 Intel Corporation
+# Copyright (c) 2020 Intel Corporation
 
-SUDO=''
-[[ $EUID -ne 0 ]] && SUDO=sudo
+PLUGIN_DIR=${PLUGINS_DIR:-"plugins"}
 
 build_bess() {
   set -e
 
-  PLUGINS=`find plugins -mindepth 1 -maxdepth 1 -type d`
+  PLUGINS=$(find "$PLUGIN_DIR" -mindepth 1 -maxdepth 1 -type d)
   echo "Found plugins: $PLUGINS"
 
-  # The BESS build script only supports specifying one --plugin at a time.
-  # First build the daemon, then rebuild for each plugin.
-  echo "Running main BESS build..."
-  ./build.py bess
-
+  CMD="./build.py -v bess"
   for PLUGIN in $PLUGINS; do
-    echo "Running plugin build: `basename $PLUGIN`..."
-    ./build.py bess --plugin $PLUGIN
+    CMD="$CMD --plugin $PLUGIN"
   done
+
+  eval $CMD
 }
 
 (return 2>/dev/null) && echo "Sourced" && return
@@ -28,7 +24,5 @@ set -o errexit
 set -o pipefail
 set -o nounset
 
-[  -z "$ENABLE_NTF" ] && exit 0
-
 echo "Building BESS..."
 build_bess
diff --git a/install_ntf.sh b/install_ntf.sh
index e109e888..d6d87edc 100755
--- a/install_ntf.sh
+++ b/install_ntf.sh
@@ -1,10 +1,11 @@
 #!/usr/bin/env bash
 # SPDX-License-Identifier: Apache-2.0
-# Copyright (c) 2019 Intel Corporation
+# Copyright (c) 2020 Intel Corporation
 
 ENABLE_NTF=${ENABLE_NTF:-""}
 NTF_COMMIT=${NTF_COMMIT:-master}
 CJOSE_COMMIT=${CJOSE_COMMIT:-9261231f08d2a3cbcf5d73c5f9e754a2f1c379ac}
+PLUGIN_DIR=${PLUGINS_DIR:-"plugins"}
 
 SUDO=''
 [[ $EUID -ne 0 ]] && SUDO=sudo
@@ -21,18 +22,20 @@ install_ntf() {
     libjansson4 \
     libtool
 
-  wget -qO cjose.zip https://github.com/cisco/cjose/archive/${CJOSE_COMMIT}.zip
+  pushd /
+  wget -qO cjose.zip "https://github.com/cisco/cjose/archive/${CJOSE_COMMIT}.zip"
   unzip cjose.zip
-  pushd cjose-${CJOSE_COMMIT}
+  rm cjose.zip
+  cd "cjose-${CJOSE_COMMIT}"
   ./configure --prefix=/usr
   make
   make install
   popd
 
-  pushd plugins
-  wget -qO ntf.zip https://github.com/Network-Tokens/ntf/archive/${NTF_COMMIT}.zip
+  wget -qO ntf.zip "https://github.com/Network-Tokens/ntf/archive/${NTF_COMMIT}.zip"
   unzip ntf.zip
-  popd
+  rm ntf.zip
+  mv "ntf-${NTF_COMMIT}" "${PLUGIN_DIR}"
 }
 
 cleanup_image() {
```
```
I0623 08:36:19.250355     1 bessd.cc:456] Loading plugin (attempt 1): /bin/modules/pcap_source.so
I0623 08:36:19.252846     1 bessd.cc:456] Loading plugin (attempt 1): /bin/modules/ntf.so
I0623 08:36:19.254974     1 bessd.cc:456] Loading plugin (attempt 1): /bin/modules/sequential_update.so
```

Reply:
this should simplify everything to a single compile

Reply:
once you commit the above, squash all commits and force push

Reply:
@krsna1729 Heya - did you test this?  I'm pretty sure multiple --plugin command line options do not work.  Also it is PLUGIN_DIR in one place and PLUGINS_DIR in another.  I'll assume PLUGIN_DIR & tidy this up.  But I think we need to build the plugins individually.  I tried this last night but it didn't work.  I'll try one more time.  If not I will leave it as it was & commit with the fixups.

Reply:
it still worked because PLUGIN_DIR was used, but yeah make it PLUGINS_DIR

Reply:
i tested it out

Reply:
I spoke too soon, it does work :)